apps/condo/domains/user/schema/ForgotPasswordService.js

```diff
@@ -133,25 +135,52 @@ const ForgotPasswordService = new GQLCustomSchema('ForgotPasswordService', {
                 if (password.length < MIN_PASSWORD_LENGTH) {
                     throw new Error(`${PASSWORD_TOO_SHORT}] Password is too short`)
                 }
-                const [action] = await ForgotPasswordActionUtil.getAll(context, {
+
+                let [action] = await ForgotPasswordActionUtil.getAll(context, {
                     token,
                     expiresAt_gte: now,
                     usedAt: null,
                 })
-
+                if (!action) {
+                    [action] = await ConfirmPhoneActionUtil.getAll(context, {
+                        token, 
+                        expiresAt_gte: now,
+                        completedAt: null,
+                    })
+                }
                 if (!action) {
                     throw new Error(`${RESET_TOKEN_NOT_FOUND}] Unable to find valid token`)
                 }
 
-                const userId = action.user.id
-                const { phone } = await getById('User', userId)
-                const tokenId = action.id
+                let phone, userId
 
-                // mark token as used
-                await ForgotPasswordActionUtil.update(context, tokenId, {
-                    usedAt: now,
-                })
+                if (typeof action.smsCode === 'number') {
```

Comment:
This is an implicit approach. The data model may change. We can add this field to ForgotPasswordAction as well.

Reply:
That's true but we are using javascript and don't have classes metainformation. I can wrap logic inside if condition but it will be complex to read, although it is temporary solution. 
See example in https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates
![изображение](https://user-images.githubusercontent.com/17229619/149207401-4c9b3114-5020-4150-9367-587cf915e310.png)

Reply:
This is true. But you can use an explicit approach. You know the model for which you found the token.

Reply:
In this case, you are trying to implicitly select a branch of the solution. It's not clear.

Reply:
Remade